--- htmlwidget/tclconfig/tcl.m4	2006-11-27 02:57:01.649152428 +0200
+++ htmlwidget/tclconfig/tcl.m4	2006-11-27 02:59:05.961945893 +0200
@@ -3648,7 +3648,12 @@
         list="`ls -d ${TCL_BIN_DIR}/../bin 2>/dev/null` \
               `ls -d ${TCL_BIN_DIR}/..     2>/dev/null` \
               `ls -d ${TCL_PREFIX}/bin     2>/dev/null`"
-        for i in $list ; do
+        for i in /usr/bin $list ; do
+            if test -f "$i/tclsh" ; then
+                REAL_TCL_BIN_DIR="`cd "$i"; pwd`"
+				TCLSH_PROG="tclsh"
+                break
+            fi
             if test -f "$i/${TCLSH_PROG}" ; then
                 REAL_TCL_BIN_DIR="`cd "$i"; pwd`"
                 break
@@ -3698,7 +3703,12 @@
         list="`ls -d ${TK_BIN_DIR}/../bin 2>/dev/null` \
               `ls -d ${TK_BIN_DIR}/..     2>/dev/null` \
               `ls -d ${TK_PREFIX}/bin     2>/dev/null`"
-        for i in $list ; do
+        for i in /usr $list ; do
+            if test -f "$i/wish" ; then
+                REAL_TK_BIN_DIR="`cd "$i"; pwd`"
+				WISH_PROG="wish"
+                break
+            fi
             if test -f "$i/${WISH_PROG}" ; then
                 REAL_TK_BIN_DIR="`cd "$i"; pwd`"
                 break
